prometheus/go_collector.go

```diff
@@ -23,6 +26,7 @@ func NewGoCollector() *goCollector {
 			"go_gc_duration_seconds",
 			"A summary of the GC invocation durations.",
 			nil, nil),
+		memStatsCollector: newMemStatsCollector(),
```

Comment:
If you really want to wrap the memstats in a different collector, then you have to call its Describe and Collect methods in the goCollector's Describe and Collect methods.